webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSIntegrationProcessorJAXWS_EJB.java

```diff
@@ -179,13 +179,18 @@ private static WebContextAnnotationWrapper getWebContextWrapper(final ClassInfo
 
         // process @PermitAll annotation
         if (webServiceClassInfo.annotations().containsKey(PERMIT_ALL_ANNOTATION)) {
-            final AnnotationInstance permitAll = webServiceClassInfo.annotations().get(PERMIT_ALL_ANNOTATION).iterator().next();
-            if (permitAll.target().equals(webServiceClassInfo)) {
-                securityRoles.add("*");
+            for (AnnotationInstance permitAll : webServiceClassInfo.annotations().get(PERMIT_ALL_ANNOTATION)) {
+                if (permitAll.target().equals(webServiceClassInfo)) {
+                    securityRoles.add("*");
+                    break;
+                }
             }
         }
-
-        return (securityRoles.size() > 0) ? Collections.unmodifiableSet(securityRoles) : Collections.<String>emptySet();
+        //if there is no class level security annotation, it will delegate to ejb's security check
+        if (securityRoles.isEmpty()) {
+            securityRoles.add("*");
```

Comment:
I don't know how this collection gets used later on, but this doesn't look right. You could have a EJB which has a @RolesAllowed annotation on a specific method and no other security annotations. In such cases, we treat the other methods as restricted (by default, unless you switch a config flag). This change, I think, is going to bypass that behaviour and instead grant access to those other methods. Like I said, I don't know how this set gets used in other places, so my comment may not be accurate.